contracts/token/Minter.sol

```diff
@@ -99,53 +148,61 @@ contract Minter is Manager, IMinter {
         return mintAmount;
     }
 
-    /*
-     * @dev Transfer tokens to a receipient
+    /**
+     * @dev Transfer tokens to a receipient. Only callable by BondingManager - always trusts BondingManager
      * @param _to Recipient address
      * @param _amount Amount of tokens
      */
-    function transferTokens(address _to, uint256 _amount) external onlyBondingManager whenSystemNotPaused {
+    function trustedTransferTokens(address _to, uint256 _amount) external onlyBondingManager whenSystemNotPaused {
         livepeerToken().transfer(_to, _amount);
     }
 
-    /*
-     * @dev Burn tokens
+    /**
+     * @dev Burn tokens. Only callable by BondingManager - always trusts BondingManager
      * @param _amount Amount of tokens to burn
      */
-    function burnTokens(uint256 _amount) external onlyBondingManager whenSystemNotPaused {
+    function trustedBurnTokens(uint256 _amount) external onlyBondingManager whenSystemNotPaused {
         livepeerToken().burn(_amount);
     }
 
-    /*
-     * @dev Withdraw ETH to a recipient
+    /**
+     * @dev Withdraw ETH to a recipient. Only callable by BondingManager or JobsManager - always trusts these two contracts
      * @param _to Recipient address
      * @param _amount Amount of ETH
      */
-    function withdrawETH(address _to, uint256 _amount) external onlyBondingManagerOrJobsManager whenSystemNotPaused {
+    function trustedWithdrawETH(address _to, uint256 _amount) external onlyBondingManagerOrJobsManager whenSystemNotPaused {
         _to.transfer(_amount);
     }
 
-    /*
-     * @dev Deposit ETH from the JobsManager
+    /**
+     * @dev Deposit ETH to this contract. Only callable by the currently registered Minter or JobsManager
      */
-    function depositETH() external payable onlyJobsManager whenSystemNotPaused returns (bool) {
+    function depositETH() external payable onlyMinterOrJobsManager whenSystemNotPaused returns (bool) {
```

Comment:
The modifier is changed to check if the caller is either the currently registered Minter or the JobsManager. The idea behind adding a check for the currently registered Minter was to restrict ETH transfer to this contract so that the contract cannot receive random ETH - it can only receive ETH from these whitelisted addresses if `depositETH` is called. So when we need to migrate to a new Minter, the old Minter is still registered with the Controller so it can transfer its ETH balance to the new Minter using `depositETH`.